gateway/mw_go_plugin_test.go

```diff
@@ -9,29 +9,45 @@ import (
 )
 
 func TestGoPluginFromTykVersion(t *testing.T) {
+	t.Parallel()
+
 	m := GoPluginMiddleware{
 		BaseMiddleware: BaseMiddleware{},
 		Path:           "",
 		SymbolName:     "test-symbol",
 	}
 
 	type testCase struct {
-		userDefinedName, inferredName string
+		version, userDefinedName, inferredName string
+	}
+	goos := runtime.GOOS
+	goarch := runtime.GOARCH
+
+	testcases := []testCase{
+		{"", "", ""},
 	}
-	os := runtime.GOOS
-	arch := runtime.GOARCH
-
-	matrix := []testCase{
-		{"plugin.so", fmt.Sprintf("./plugin_%v_%v_%v.so", VERSION, os, arch)},
-		{"/some/path/plugin.so", fmt.Sprintf("/some/path/plugin_%v_%v_%v.so", VERSION, os, arch)},
-		{"/some/path/plugin", fmt.Sprintf("/some/path/plugin_%v_%v_%v.so", VERSION, os, arch)},
-		{"./plugin.so", fmt.Sprintf("./plugin_%v_%v_%v.so", VERSION, os, arch)},
-		{"", ""},
+
+	// Go middleware compiler only reads the pre-injected
+	// VERSION value from version.go, expecting to search
+	// plugin with clean version in filename (no -rc16).
+	expectVersion := "v4.1.0"
```

Comment:
@titpetric can we inject this value from `version.go`?

Reply:
VERSION in version.go is unsanitized, so not exactly; it would add a coupling to a value, which doesn't need to be part of the test. Here all inputs are self-contained in the test. Explained in the comment above ;)

Reply:
@titpetric I understand that it would add a coupling to a value - in a test. Maybe I'm missing something but this test is going to fail whenever we update `VERSION` in `version.go` in the future?

Reply:
nevermind, got it 👍